src/CodeGen_Hexagon.cpp

```diff
@@ -1810,6 +1818,16 @@ void CodeGen_Hexagon::visit(const Call *op) {
                                 type_suffix(op->args[1], op->args[2], false),
                                 op->args);
             return;
+        } else if (op->is_intrinsic(Call::abs)) {
+            internal_assert(op->args.size() == 1);
+            Type ty = op->args[0].type();
+            if ((ty.is_vector() && ty.is_int())) {
+                if ((ty.bits() == 8 && (target.features_any_of({Target::HVX_v65, Target::HVX_v66})))
+                || (ty.bits() == 16 || ty.bits() == 32)) {
+                    value = call_intrin(op->type, "halide.hexagon.abs" + type_suffix(op->args[0]), op->args);
+                    return;
+                }
+            }
```

Comment:
Changed this code to handle only what it can handle and fall through for failure case.